### PR TITLE
Fix set 2d color no working

### PR DESCRIPTION
**Pull request by singlecoder**
## Summary by CodeRabbit

- **New Features**
  - Added color change handling to `SpriteRenderer` and `TextRenderer` classes for more dynamic rendering.

- **Bug Fixes**
  - Improved color update logic for `SpriteRenderer` and `TextRenderer` to address issues with color rendering.

- **Tests**
  - Updated tests for `SpriteRenderer` to validate individual RGBA components, ensuring more precise color validation.

### CODE REVIEW

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

File: tests/src/core/SpriteRenderer.test.ts [hunk 1]
```diff
@@ -76,7 +76,10 @@ describe("SpriteRenderer", async () => {
     spriteRenderer.color.set(1, 0, 0, 1);
     expect(Color.equals(spriteRenderer.color, new Color(1, 0, 0, 1))).to.eq(true);
     spriteRenderer.color = new Color(0, 1, 0, 1);
-    expect(spriteRenderer.color).to.deep.eq(new Color(0, 1, 0, 1));
+    expect(spriteRenderer.color.r).to.eq(0);
+    expect(spriteRenderer.color.g).to.eq(1);
+    expect(spriteRenderer.color.b).to.eq(0);
+    expect(spriteRenderer.color.a).to.eq(1);
```
Comment (cptbtptpbcptdtptp) [inline tests/src/core/SpriteRenderer.test.ts:82]: It is best to check dirty to determine whether it takes effect.

File: packages/core/src/2d/sprite/SpriteRenderer.ts [hunk 2]
```diff
@@ -375,7 +376,8 @@ export class SpriteRenderer extends Renderer {
 
     super._onDestroy();
 
-    this._entity = null;
+    //@ts-ignore
+    this._color._onValueChanged = null;
```
Comment (GuoLei1990) [inline packages/core/src/2d/sprite/SpriteRenderer.ts:380]: Don't need

File: packages/core/src/2d/text/TextRenderer.ts [hunk 3]
```diff
@@ -314,6 +310,10 @@ export class TextRenderer extends Renderer {
 
     super._onDestroy();
 
+    //@ts-ignore
+    this._color._onValueChanged = null;
```
Comment (GuoLei1990) [inline packages/core/src/2d/text/TextRenderer.ts:314]: Don't need

**Review by GuoLei1990 — APPROVED** [review]

**Pull request merged**